### ISSUE TITLE
want to know about volumentry data

### ISSUE THREAD

**Issue by monicakumari** [atolcd/alfresco-audit-share#20]
Hello,
I was testing this add-on, everything seems working fine. but I am having some issue with volumentry data. It is showing the data for volumentry for 4 hours - 4am to 6 am and 6am to 8am. attached screenshots for the same. 

![screenshot from 2018-04-16 14-41-54](https://user-images.githubusercontent.com/26789477/38801401-56c05158-4187-11e8-98e2-1ffe0e57b1b0.png)
![screenshot from 2018-04-16 14-42-13](https://user-images.githubusercontent.com/26789477/38801402-56eac154-4187-11e8-8584-a232f48f1df5.png)
![screenshot from 2018-04-16 14-42-27](https://user-images.githubusercontent.com/26789477/38801403-5716102a-4187-11e8-9187-287d3d1a8980.png)

I just wanted to know the working and use of "volumentry" action. like, if it is showing the volume for the sites and I want to see the volume right now just to know, what is the current volume of a site, how would I know.
just having some questions like this. 

Thanks.

**Comment by monicakumari** [comment]
I overrided share.stats.sites.volumetry.trigger property. 
share.stats.sites.volumetry.trigger=0 0/1 0 * * ? instead of share.stats.sites.volumetry.trigger=0 0 0/4 * * ?.
but this is not working. changed this property inside the file itself and also in alfresco-global.properties file.
I think scheduler is not triggering.

**Comment by bforest** [comment]
Hi,

On the "day" view, data are always displayed for 2 hours range.
You can modified CRON expression to increase or decrease the job frequency but it will not changed the chart Y axis displaying.

Regards.

**Comment by monicakumari** [comment]
Thank you for your reply. 
I understood this. On the "day" view, data are always displayed for 2 hours range. 
But I wanted to know,why it is showing the same time everyday, i.e, 4am to 6am and 6am to 8am.?

I have tried this, I overrided share.stats.sites.volumetry.trigger property.
share.stats.sites.volumetry.trigger=0 0/5 0 * * ? instead of share.stats.sites.volumetry.trigger=0 0 0/4 * * ?.
changed this property inside the file itself and also in alfresco-global.properties file but this is not working.
I think scheduler is not triggering.

could you please help me with this, I want this scheduler to be triggered in every 5 minutes. Is this possible ? and Is it possible to show correct time on x-axis when the scheduler triggers ?

**Comment by bforest** [comment]
Hi,

By default, volumetry job is launched every 4 hours : share.stats.sites.volumetry.trigger=0 0 0/4 * * ? (https://www.freeformatter.com/cron-expression-generator-quartz.html)

![volumetry](https://user-images.githubusercontent.com/2580135/38916213-caff8a42-42e6-11e8-95d4-9f57dcd89a1a.png)

For testing, I've modified the cron expression to: 0 */5 * ? * * (every 5 minutes).
Do to that, you need to edit the alfresco-global.properties file of your Alfresco and **you must restart Tomcat**.

You can enable loggers to see when job is executed:
log4j.logger.com.atolcd.alfresco.web.scripts.shareStats.InsertAuditPost=DEBUG
log4j.logger.com.atolcd.alfresco.repo.jscript.ShareStats=DEBUG

As expected, data are inserted every 5 minutes:
```
2018-04-18 08:40:02,505 DEBUG [com.atolcd.alfresco.repo.jscript.ShareStats] [DefaultScheduler_Worker-4]   -- Site Volumetry swsdp: 57
2018-04-18 08:40:02,543 INFO  [com.atolcd.alfresco.web.scripts.shareStats.InsertAuditPost] [DefaultScheduler_Worker-4] Volumetry entry successfully inserted.

2018-04-18 08:45:00,274 DEBUG [com.atolcd.alfresco.repo.jscript.ShareStats] [DefaultScheduler_Worker-4]   -- Site Volumetry swsdp: 57
2018-04-18 08:45:00,308 INFO  [com.atolcd.alfresco.web.scripts.shareStats.InsertAuditPost] [DefaultScheduler_Worker-4] Volumetry entry successfully inserted.

2018-04-18 08:47:42,103 INFO  [com.atolcd.alfresco.web.scripts.shareStats.InsertAuditPost] [http-bio-8080-exec-25] Entry successfully inserted: {"auditActionName":"file-added","auditUserId":"admin","auditAppName":"document","auditTime":"1524034062092","id":"84","auditObject":"workspace://SpacesStore/981a49d1-36e6-4cb1-b78e-3f420050950c","auditSite":"swsdp","auditNodeType":"cm:content"}

2018-04-18 08:50:00,260 DEBUG [com.atolcd.alfresco.repo.jscript.ShareStats] [DefaultScheduler_Worker-5]   -- Site Volumetry swsdp: 58
2018-04-18 08:50:00,283 INFO  [com.atolcd.alfresco.web.scripts.shareStats.InsertAuditPost] [DefaultScheduler_Worker-5] Volumetry entry successfully inserted.

2018-04-18 08:55:00,280 DEBUG [com.atolcd.alfresco.repo.jscript.ShareStats] [DefaultScheduler_Worker-6]   -- Site Volumetry swsdp: 58
2018-04-18 08:55:00,303 INFO  [com.atolcd.alfresco.web.scripts.shareStats.InsertAuditPost] [DefaultScheduler_Worker-6] Volumetry entry successfully inserted.
```

Regards.

**Comment by monicakumari** [comment]
Thank you so much for your help. I'll try this and let you know. 

Thanks..

**Comment by monicakumari** [comment]
yes, It is triggering now but data is not updating on UI(chart). Its is still showing the data same as when the tomcat started. after that, I uploaded more douments, and updated data I can see on console logs but not updating on UI side. bar chart/line chart is still showing the same size.

Thanks.

**Comment by bforest** [comment]
Please, can you confirm that your Alfresco is running all the day?

Regards.

**Comment by bforest** [comment]
Ok, it's due to the query: https://github.com/atolcd/alfresco-audit-share/blob/master/module_alfresco/src/main/config/ibatis/org.hibernate.dialect.Dialect/share-stats-common-SqlMap.xml#L540 because we use **max** function.

Regards.

**Comment by monicakumari** [comment]
ohh .. okay.. Thank you for your quick reply.  so, what should I use instead ?. to make it update data on graph, even in case of deletion ?

Thanks.

**Comment by bforest** [comment]
**min** or **avg**, I suppose (https://www.w3schools.com/sql/sql_min_max.asp / https://www.w3schools.com/sql/sql_count_avg_sum.asp).

Regards.

**Comment by monicakumari** [comment]
If I use 
select
     site_size

instead of 

select
max(site_size)

will it work ??
